src/steps/pipeline/index.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+
+import { Entities, Steps } from '../constants';
+import { createPipelineEntity } from './converter';
+
+export async function fetchPipelines({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await jobState.iterateEntities(
+    { _type: Entities.USER_GROUP._type },
+    async (userGroupEntity) => {
+      if (userGroupEntity.slug) {
+        await apiClient.iteratePipelines(
+          userGroupEntity.slug.toString(),
+          async (pipeline) => {
+            const pipelineEntity = createPipelineEntity(pipeline);
+            await jobState.addEntity(pipelineEntity);
+          },
+        );
+      }
+    },
+  );
+}
+
+export const pipelinesSteps: IntegrationStep<IntegrationConfig>[] = [
+  {
+    id: Steps.PIPELINES,
+    name: 'Fetch Pipelines',
+    entities: [Entities.CONFIGURATION],
+    relationships: [],
+    dependsOn: [Steps.ACCOUNT, Steps.USER_GROUPS],
```

Comment:
The pipeline step only depends on USER_GROUPS step. It's true that USER_GROUPS step in turns depends on ACCOUNT step, but we don't need to put a dependency for that here since USER_GROUPS will wait for it on "its own".

```suggestion
    dependsOn: [Steps.USER_GROUPS],
```